File: e2e/go/strategy_test.go
// E2e tests for category: strategy
package e2e_test

import (
	"strings"
	"testing"

	pkg "github.com/kreuzberg-dev/kreuzcrawl"
)

func Test_StrategyBestFirstSeed(t *testing.T) {
	// BestFirst strategy always processes the seed URL first
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'crawl.pages_crawled' not available on result type
	// skipped: field 'strategy.first_page_url_contains' not available on result type
}

func Test_StrategyBfsDefaultOrder(t *testing.T) {
	// BFS strategy visits pages in breadth-first order
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'crawl.pages_crawled' not available on result type
	// skipped: field 'strategy.crawl_order' not available on result type
}

func Test_StrategyDfsDepthFirst(t *testing.T) {
	// DFS strategy visits pages in depth-first order
	engine, createErr := pkg.CreateEngine()
	if createErr != nil {
		t.Fatalf("create handle failed: %v", createErr)
	}
	_, err := pkg.Scrape(engine, "")
	if err != nil {
		t.Fatalf("call failed: %v", err)
	}
	// skipped: field 'crawl.pages_crawled' not available on result type
	// skipped: field 'strategy.crawl_order' not available on result type
}

File: e2e/rust/tests/strategy_test.rs
//! E2e tests for category: strategy

use kreuzcrawl::scrape;
use kreuzcrawl::create_engine;

#[tokio::test]
async fn test_strategy_best_first_seed() {
    // BestFirst strategy always processes the seed URL first
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'crawl.pages_crawled' not available on result type
    // skipped: field 'strategy.first_page_url_contains' not available on result type
}

#[tokio::test]
async fn test_strategy_bfs_default_order() {
    // BFS strategy visits pages in breadth-first order
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'crawl.pages_crawled' not available on result type
    // skipped: field 'strategy.crawl_order' not available on result type
}

#[tokio::test]
async fn test_strategy_dfs_depth_first() {
    // DFS strategy visits pages in depth-first order
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let _ = scrape(&engine, &url).await.expect("should succeed");
    // skipped: field 'crawl.pages_crawled' not available on result type
    // skipped: field 'strategy.crawl_order' not available on result type
}
